src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -85,9 +85,13 @@ public void Start(int maximumProcessingConcurrency, CancellationToken token)
                 WaitTimeSeconds = 20,
                 AttributeNames = new List<string> { "SentTimestamp" },
                 MessageAttributeNames = new List<string> { "*" },
-                VisibilityTimeout = configuration.MessageVisibilityTimeout,
             };
 
+            if (configuration.MessageVisibilityTimeout.HasValue)
```

Comment:
As discussed with @danielmarbach, it's important to keep in mind that this check might be misleading as it doesn't express the fact that the `MessageVisibilityTimeout` property is only set when using the pub/sub compatibility mode. It might be worth either documenting with a comment why the if exists or adding an additional condition like (pseudo-code):

```
if (compatibilityModeIsEnabled && configuration.MessageVisibilityTimeout.HasValue)
```

Reply:
I would prefer having a comment here. Otherwise good to go